### PR TITLE
Adding MaxProj as a cli filter command

### LINKED ISSUE #659: CLI should have ability to compute a max projection
There are cases in which one filters a full 3D stack, takes a maximum projection across Z, then does all re-maining processing steps (e.g., spot-detection, segmentation, decoding, etc.) on the resulting 2D stack.

This is easy in the API, modulo the fact that, right now, ImageStack.max_proj does not require an ImageStack. But you can't do it in the CLI easily.

### PR DESCRIPTION
**Pull request by shanaxel42**
Added a small wrapper class around Imagestack.max_proj so that it can be called from the command line. ex. `starfish filter -i input_file -o outout_file MaxProj -dims r -dims c`

fixes #659

### CODE REVIEW

File: starfish/image/_filter/max_proj.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    def run(
+            self, stack: ImageStack, in_place: bool = False, verbose: bool = False,
+            n_processes: Optional[int] = None
+    ) -> ImageStack:
+        return stack.max_proj(*tuple(Indices(dim) for dim in self.dims))
+
+    @staticmethod
+    @click.command("MaxProj")
+    @click.option("-dims", type=str, multiple=True,
```
[the remark this answers is not in the document]
Comment (shanaxel42) [reply starfish/image/_filter/max_proj.py]: ah yea that seems right.

File: starfish/image/_filter/max_proj.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    def run(
+            self, stack: ImageStack, in_place: bool = False, verbose: bool = False,
+            n_processes: Optional[int] = None
+    ) -> ImageStack:
+        return stack.max_proj(*tuple(Indices(dim) for dim in self.dims))
+
+    @staticmethod
+    @click.command("MaxProj")
+    @click.option("--dims", type=str, multiple=True,
+                  help="The dimensions the Imagestack should max project over")
```
Comment (ttung) [inline starfish/image/_filter/max_proj.py]: It would be nice to have the list of dimensions available here.

Comment (joshmoore) [reply starfish/image/_filter/max_proj.py]: If so, I'd suggest refactoring `StarfishIndex` to `starfish.util.click`

File: starfish/image/_filter/max_proj.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            self, stack: ImageStack, in_place: bool = False, verbose: bool = False,
+            n_processes: Optional[int] = None
+    ) -> ImageStack:
+        return stack.max_proj(*tuple(Indices(dim) for dim in self.dims))
+
+    @staticmethod
+    @click.command("MaxProj")
+    @click.option("--dims", type=str, multiple=True,
+                  help="The dimensions the Imagestack should max project over. Options:"
+                       "(r, c, z, x, or y) For multiple dimensions add multiple --dims. Ex."
```
Comment (dganguli) [inline starfish/image/_filter/max_proj.py]: (r, c, z, y, x)

**Pull request merged**